src/OpenSpaceToolkit/Core/Error/RuntimeError.cpp

```diff
@@ -12,19 +11,19 @@ namespace error
 RuntimeError::RuntimeError(const String& aMessage)
     : Exception(String::Empty()),
       message_(aMessage),
-      stackTrace_(boost::stacktrace::to_string(boost::stacktrace::stacktrace())),
-      what_(stackTrace_ + message_)
+      stackTrace_(String::SanitizeUTF8(boost::stacktrace::to_string(boost::stacktrace::stacktrace()))),
+      what_(aMessage + "\n" + stackTrace_)
```

Comment:
Doesn't Python do something like:
`aMessage + "\n" + stackTrace_ + "\n" + aMessage`

Think there's value in doing that while we're switching it around anyway?
It's sometimes nice to have the message at the bottom when the terminal auto-scrolls, so you don't have to scroll up to see it.

Reply:
Python does:
Exception -> Message
Where message used to be stackTrace_ + message_ (no new line)

New format is:
```
RuntimeError: {Directory} is undefined.
 0# ostk::core::error::RuntimeError::RuntimeError(ostk::core::types::String const&) at /app/src/OpenSpaceToolkit/Core/Error/RuntimeError.cpp:14
 1# ostk::core::error::runtime::Undefined::Undefined(ostk::core::types::String const&) at /app/src/OpenSpaceToolkit/Core/Error/Runtime/Undefined.cpp:15
 2# ostk::core::fs::Directory::exists() const [clone .cold] at /app/src/OpenSpaceToolkit/Core/FileSystem/Directory.cpp:56
 3# 0x00007FAD25C90B33 in /opt/conda/lib/python3.11/site-packages/ostk/core/OpenSpaceToolkitCorePy.cpython-311-x86_64-linux-gnu.so
 4# 0x00007FAD25CC0DF4 in /opt/conda/lib/python3.11/site-packages/ostk/core/OpenSpaceToolkitCorePy.cpython-311-x86_64-linux-gnu.so
 5# 0x00005630E38E1DC6 in /opt/conda/bin/python3.11
 6#  in /opt/conda/bin/python3.11
 7# 0x00005630E38CE992 in /opt/conda/bin/python3.11
 8# 0x00005630E3986D36 in /opt/conda/bin/python3.11
 9# arams in /opt/conda/bin/python3.11
10# 0x00005630E399DBAE in /opt/conda/bin/python3.11
11# 0x00005630E38DB6CF in /opt/conda/bin/python3.11
12# _reset in /opt/conda/bin/python3.11
13# 0x00005630E38CE992 in /opt/conda/bin/python3.11
14# 0x00005630E399AA79 in /opt/conda/bin/python3.11
15# 0x00005630E38D1580 in /opt/conda/bin/python3.11
16# 0x00005630E399AA79 in /opt/conda/bin/python3.11
17# 0x00005630E38D1580 in /opt/conda/bin/python3.11
18# 0x00005630E399AA79 in /opt/conda/bin/python3.11
19# 0x00005630E399B647 in /opt/conda/bin/python3.11
20# L in /opt/conda/bin/python3.11
21# _reset in /opt/conda/bin/python3.11
22# 0x00005630E38CE992 in /opt/conda/bin/python3.11
23# 
     in /opt/conda/bin/python3.11
24# 0x00005630E3911673 in /opt/conda/bin/python3.11
25# newdata in /opt/conda/bin/python3.11
26# 0x00005630E38D24E7 in /opt/conda/bin/python3.11
27# 0x00005630E3986D36 in /opt/conda/bin/python3.11
28# arams in /opt/conda/bin/python3.11
29# 0x00005630E39A4F2A in /opt/conda/bin/python3.11
30# 0x00005630E39A1343 in /opt/conda/bin/python3.11
31# 0x00005630E39B6300 in /opt/conda/bin/python3.11
32# 0x00005630E39B5C5E in /opt/conda/bin/python3.11
33#  in /opt/conda/bin/python3.11
34# I in /opt/conda/bin/python3.11
35# PublicKeyInfo_der_does_selection in /opt/conda/bin/python3.11
36# 0x00007FAD29724D90 in /lib/x86_64-linux-gnu/libc.so.6
37# __libc_start_main in /lib/x86_64-linux-gnu/libc.so.6
38# 0x00005630E3974E3D in /opt/conda/bin/python3.11
```

IMO this is cleaner because we can see the Exception: -> Message and then the stack trace is below that.
Mostly because I think the order of importance of information is Exception Type, Message, Stacktrace.

Reply:
I can swap it around if we prefer that, but the new line is definitely required from what I can tell.

Reply:
Yeah I agree newline is very nice. 

Maybe it's a Jupyter vs. terminal thing. In jupyter, you tend to see the top of the error message first, but in terminal you tend to see the bottom first.

I personally prefer error message on the bottom because it feels more optimized for terminal, but no super strong opinions. I see the continuity-niceness in your current approach.

Reply:
Yeah that's a fair point on jupyter vs terminal. Maybe we can use this for a bit and if it's a pita we can have the message both at the top and the bottom :P